integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/jms/inbound/transport/test/JMSInboundTransactionTestCase.java

```diff
@@ -26,121 +25,118 @@
-		try {
-			consumer.connect("localq");
-			poppedMessage = consumer.popMessage();
-		} finally {
-			consumer.disconnect();
-		}
-		if (status.equals("Committed") && poppedMessage == null) {
-			return true;
-		} else if (status.equals("Rollbacked") && poppedMessage != null) {
-			return true;
-		}
-		return false;
-	}
-
-	@AfterClass(alwaysRun = true)
-	public void destroy() throws Exception {
-		super.cleanup();
-		Thread.sleep(3000);
-		activeMQServer.stopJMSBroker();
-
-	}
+    private LogViewerClient logViewerClient = null;
+    private ActiveMQServer activeMQServer = new ActiveMQServer();
+    String message;
+
+    @BeforeClass(alwaysRun = true)
+    protected void init() throws Exception {
+
+        activeMQServer.startJMSBroker();
+        super.init();
+
+        verifySequenceExistence("jmsInboundEpTransactionRollbackTestInSequence");
+        verifySequenceExistence("jmsInboundEpTransactionRollbackTestFaultSequence");
+        verifySequenceExistence("jmsInboundEndpointTransactionRequestHandlerSequence");
+
+        message = String.valueOf(esbUtils.loadResource("artifacts/ESB/jms/inbound/transport/message.xml"));
+        logViewerClient = new LogViewerClient(contextUrls.getBackEndUrl(), getSessionCookie());
+    }
+
+    /**
+     * Tests whether committing a transaction removes the message from the queue.
+     *
+     * @throws Exception for any unexpected exception
+     */
+    @Test(groups = { "wso2.esb" },
+          description = "Successfully committing the message")
+    public void testTransactionCommit() throws Exception {
+
+        String queueName = "testTransactionCommitQueue";
+        JMSQueueMessageProducer sender = new JMSQueueMessageProducer(JMSBrokerConfigurationProvider.getInstance()
+                                                                                                   .getBrokerConfiguration());
+
+        logViewerClient.clearLogs();
+
+        try {
+            sender.connect(queueName);
+            sender.pushMessage(message);
+        } finally {
+            sender.disconnect();
+        }
+
+        addInboundEndpoint(esbUtils.loadResource(
+                "artifacts/ESB/jms/inbound/transport/jmsInboundTransactionCommitInboundEp.xml"));
+        // Wait till the message is picked from inbound endpoint
+        Thread.sleep(8000);
+
+        Assert.assertTrue(Utils.assertIfSystemLogContains(logViewerClient, "Committed"),
+                          "Did not find the \"Committed\" log");
+        deleteInboundEndpoints();
+
+        Assert.assertTrue(isQueueEmpty(queueName), "Queue (" + queueName + ") should be empty after commit");
+    }
+
+    /**
+     * Tests whether rolling back a transaction does not removes the message from the queue.
+     *
+     * @throws Exception
+     */
+    @Test(groups = { "wso2.esb" },
+          description = "Rolling back the failed message to the queue",
+          enabled = true)
```

Comment:
remove enable true. It is by default enabled.

Reply:
:+1: